device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1159,7 +1159,8 @@ private void reconnectDeviceSession(TransportException transportException, Strin
         RetryDecision retryDecision = null;
 
         while (this.deviceConnectionStates.get(deviceId) == IotHubConnectionStatus.DISCONNECTED_RETRYING
-                && !hasReconnectOperationTimedOut)
+                && !hasReconnectOperationTimedOut
+                && transportException.isRetryable())
```

Comment:
We abandon reconnection on unretryable errors for non multiplexed client cases, so I consider it a bug that we don't do the same for multiplexed cases.